# HG changeset patch
# User Betterbird <betterbird@betterbird.eu>
# Date 1673388995 -3600
# Parent  38f467b2bfad6d554c297e3a34ff3d98898de31d
Fix logic in MailNotificationManager.jsm: Not showing badge led to not showing tooltip :-(

diff --git a/mailnews/base/src/MailNotificationManager.jsm b/mailnews/base/src/MailNotificationManager.jsm
--- a/mailnews/base/src/MailNotificationManager.jsm
+++ b/mailnews/base/src/MailNotificationManager.jsm
@@ -432,28 +432,29 @@ class MailNotificationManager {
     this._updatingUnreadCount = true;
 
     this._logger.debug(
       `Update unreadMailCount=${this._unreadMailCount}, unreadChatCount=${this._unreadChatCount}`
     );
     let count = this._unreadMailCount + this._unreadChatCount;
     let tooltip = "";
     if (AppConstants.platform == "win") {
-      if (!Services.prefs.getBoolPref("mail.biff.show_badge", true)) {
-        count = 0;
-      }
       if (count > 0) {
         tooltip = await l10n.formatValue("unread-messages-os-tooltip", {
           count,
         });
         if (this._unreadMailExtra) {
           tooltip += this._unreadMailExtra;
         }
       }
-      await WinUnreadBadge.updateUnreadCount(count, tooltip);
+      if (Services.prefs.getBoolPref("mail.biff.show_badge", true)) {
+        await WinUnreadBadge.updateUnreadCount(count, tooltip);
+      } else {
+        await WinUnreadBadge.updateUnreadCount(0, "");
+      }
     }
     this._osIntegration?.updateUnreadCount(count, tooltip);
 
     this._updatingUnreadCount = false;
     if (this._pendingUpdate) {
       // There was at least one _updateUnreadCount call while we were rendering
       // the badge. Render one more time will ensure the badge reflects the
       // current state.
